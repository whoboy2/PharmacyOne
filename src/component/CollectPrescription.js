import { Button, TextField } from '@material-ui/core';
import React, {Component, Fragment} from 'react';
import { collectMeds } from '../helpers/firestore-helpers/database';
import { withStyles  } from '@material-ui/core/styles';
import PropTypes from 'prop-types';

const styles  = (theme) => ({
    button: {
        color: 'white',
        backgroundColor: '#2C6D6A',
    
        '&:hover': {
            color: 'black'
        },
    },    
  });

class CollectPrescription extends Component {
    constructor(props){
        super(props);
        this.handleSubmit = this.handleSubmit.bind(this);
    }
    
    // Function to handle submission of the form
    handleSubmit = (event) => {
        event.preventDefault();
        collectMeds(event.target.pres_id.value);
    }
    render() {
        const { classes } = this.props;
        return (
            <Fragment>
                <form onSubmit={this.handleSubmit}>
                    <TextField
                        variant="outlined"
                        margin="dense"
                        required
                        fullWidth
                        id="pres_id"
                        label="Prescription ID"
                        name="pres_id"
                        autoFocus />
                    <Button
                        className={classes.button}
                        id="submit-button"
                        type="submit"
                        variant="contained">
                        Mark Collected
                    </Button>
                </form>

            </Fragment>
        )
    }
}

CollectPrescription.propTypes = {
    classes: PropTypes.object.isRequired,
};


export default withStyles(styles)(CollectPrescription)