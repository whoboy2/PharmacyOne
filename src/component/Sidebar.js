import React from 'react'
import HomeIcon from '@material-ui/icons/Home';
import LocalHospitalIcon from '@material-ui/icons/LocalHospital';
import DescriptionIcon from '@material-ui/icons/Description';
import PeopleIcon from '@material-ui/icons/People';
import ExitToAppIcon from '@material-ui/icons/ExitToApp';
import ReceiptIcon from '@material-ui/icons/Receipt';
import logo1 from '../images/p1logo.png';

export const SidebarData = [
    {
        title:"Home",
        icon: <HomeIcon />,
        link: "/home"
    },

    {
        title:"Blood Work",
        icon: <DescriptionIcon />,
        link: "/bloodwork"
    },

    {
        title:"Medication",
        icon: <LocalHospitalIcon />,
        link: "/medication"
    },

    {
        title:"Prescriptions",
        icon: <ReceiptIcon />,
        link: "/prescriptions"
    },

    {
        title:"Patient",
        icon: <PeopleIcon />,
        link: "/patient"
    },

    {
        title:"Log Out",
        icon: <ExitToAppIcon />,
        link: "/"
    }
];

function Sidebar() {
    return (
        <div className="sidebar">
            <div className="title">
                <img src={ logo1 } alt='Pharmacy One logo' className="logo"/>
                <h1>PharmacyOne</h1>
            </div>
            <ul className="sidebarList">
                {SidebarData.map((val, key) => {
                    return (
                            <li 
                                key={key}
                                className="row"
                                id={window.location.pathname === val.link ? "active" : ""}
                                onClick={() => {
                                   window.location.pathname = val.link;
                                }}
                            >
                                <div id="icon">{val.icon}</div><div id="title">{val.title}</div>
                            </li>
                    )
                })}
            </ul>
        </div>
    )
}

export default Sidebar