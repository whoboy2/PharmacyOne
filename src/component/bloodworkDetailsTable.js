import React, {Component} from 'react';
const { bloodworkDetails }
  = require("../helpers/firestore-helpers/database");

class bloodworkDetailsTable extends Component {
    componentDidMount() {
        var details="";
        bloodworkDetails
        // .limit(20)
        .orderBy("bloodWorkID")
        .get()
        .then(data => {
            data.forEach((doc) => {
            // doc.data() is never undefined for query doc snapshots
            details += "<tr>" 
            + "<td>" + doc.data().bloodWorkID +"</td>"
            + "<td>" + doc.data().bloodWorkCode +"</td>"
            + "<td>" + doc.data().bloodWorkName +"</td>"
            + "</tr>"
            })
            document.getElementById('bloodwork-details').innerHTML += details;
        });
    }

    render() {
        return (
            <div className="bloodwork-detail-content">
                <h2>Blood Work Details</h2>
                <table className="table" id="bloodwork-details">
                    <tbody>
                        <tr>
                            <th>Blood Work ID</th>
                            <th>Blood Work Code</th>
                            <th>Blood Work Name</th>
                        </tr>
                    </tbody>
                </table>
            </div>
        )
    }
}

export default bloodworkDetailsTable