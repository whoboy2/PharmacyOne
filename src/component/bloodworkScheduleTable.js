import React, {Component} from 'react';
const { bloodworkSchedule }
  = require("../helpers/firestore-helpers/database");

function getLastMonday() {
  var date = new Date();
    var day = date.getDay();
    var prevMonday = new Date();
    if(date.getDay() === 0){
        prevMonday.setDate(date.getDate() - 7);
    }
    else {
        prevMonday.setDate(date.getDate() - (day-1));
    }
    return prevMonday;
}

function getNextMonday() {
  var date = new Date();
    //var day = date.getDay();
    var nextMonday = new Date();
    if(date.getDay() === 0){
        nextMonday.setDate(date.getDate() + 14);
    }
    else {
        nextMonday.setDate(date.getDate() + 14);
    }
    return nextMonday;
}

function formatDate(date, format) {
  var d = new Date(date),
      month = '' + (d.getMonth() + 1),
      day = '' + d.getDate(),
      year = d.getFullYear();
      if (year > 2050) year -= 1969;

  if (month.length < 2) 
      month = '0' + month;
  if (day.length < 2) 
      day = '0' + day;

  if (format === 'mmddyyy') return [month, day, year].join('/');
  else return [day, month, year].join('/');
}

class bloodworkScheduleTable extends Component {
    componentDidMount() {
        var schedule = "";
        // Currently limiting the number of patients returned to 20.
        var monday = getLastMonday();
        var nextMon = getNextMonday();

        let n=0;
        bloodworkSchedule
          .limit(20)
          .orderBy("bloodworkDue")
          .get()
          .then(data => {
            data.forEach((doc) => {
              var date = new Date(doc.data().bloodworkDue * 1000);
              date = formatDate(date, 'mmddyyy');
              date = new Date(date);
              // console.log(date, monday, nextMon)
              
              if ( ( (date.getTime() > monday.getTime()) && (date.getTime() < nextMon.getTime()) && (doc.data().completed === false))
                || 
              ( (date.getTime() < monday.getTime()) && (doc.data().completed === false))) {

                n+=1;
                schedule += "<tr>" 
                + "<td>" + doc.data().NHS_ID +"</td>"
                + "<td>" + doc.data().patientName +"</td>"
                + "<td>" + doc.data().medication +"</td>"
                + "<td>" + doc.data().testType +"</td>"
                + "<td>" + formatDate(date, 'ddmmyyy') +"</td>"
                + "<td>" + doc.data().completed +"</td>"
      
                + "</tr>"
              }
            })
            document.getElementById('bloodwork-schedule').innerHTML += schedule;
            document.getElementById('bloodwork-button').innerHTML = n ;
        });
    }

    render() {
        return (
        <div>
          <h2>Patients needing bloodwork</h2>
          <table className="table" id="bloodwork-schedule">
            <tbody>
              <tr>
                <th>NHS ID</th>
                <th>Name</th>
                <th>Medication</th>
                <th>Type of Blood Test</th>
                <th>Bloodwork Expiry</th>
                <th>Completed?</th>
              </tr>
            </tbody>
          </table>
        </div>  
        )
    }
}

export default bloodworkScheduleTable