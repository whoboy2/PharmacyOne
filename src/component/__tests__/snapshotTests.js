import PatientTable from "../patientsTable.js";
import PrescriptionTable from "../prescriptionTable.js"
import bloodworkDetailsTable from "../bloodworkDetailsTable"
import bloodworkScheduleTable from "../bloodworkScheduleTable"
import AllPrescriptionTable from "../allPrescriptionsTable"
import Sidebar from '../Sidebar'
import BloodworkStatusButton from '../bloodworkStatusButton'
import MedicationStatusButton from '../medicationStatusButton'
import CollectPrescription from "../CollectPrescription"

const { allPatients } = require("../../helpers/firestore-helpers/database");

import Enzyme from "enzyme";
import {shallow, render, mount} from "enzyme";
import toJson from 'enzyme-to-json';
import Adapter from "@wojtekmaj/enzyme-adapter-react-17"
Enzyme.configure({adapter: new Adapter()})

describe('renderSnapshots', () => {
    it('rendersPatientTable', () => {
      const wrapper = shallow(<PatientTable/>);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('rendersPrescriptionTable', () => {
      const wrapper = shallow(<PrescriptionTable/>);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renderSidebar', () => {
      const wrapper = shallow(<Sidebar/>);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renderBloodButton', () => {
      const wrapper = shallow(<BloodworkStatusButton/>);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renderMedButton', () => {
      const wrapper = shallow(<MedicationStatusButton/>);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renderCollectPrescription', () => {
      const wrapper = shallow(<CollectPrescription/>);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renderBloodworkDetailsTable', () => {
      const wrapper = shallow(<bloodworkDetailsTable/>);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renderBloodworkScheduleTable', () => {
      const wrapper = shallow(<bloodworkScheduleTable/>);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renderAllPrescriptionTable', () => {
      const wrapper = shallow(<AllPrescriptionTable/>);
      expect(toJson(wrapper)).toMatchSnapshot();
    });


  });

describe('DB Connection', () => {
    it('checkAllPatientsConnection', () => {
      var pateintsArray = allPatients.where("NHS_ID","==","7091277674");
      var value = "";
      pateintsArray.get().then(data => {
          data.forEach(doc => {
              value += doc.data().name;
          })
      })
      expect(value).toEqual("");
    });

  });
