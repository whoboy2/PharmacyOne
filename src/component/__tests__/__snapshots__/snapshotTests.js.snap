// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderSnapshots renderAllPrescriptionTable 1`] = `
<div>
  <h2>
    All prescriptions
  </h2>
  <table
    className="table"
    id="prescription-table"
  >
    <tbody>
      <tr>
        <th>
          Prescription No.
        </th>
        <th>
          Name
        </th>
        <th>
          NHS ID
        </th>
        <th>
          Medication
        </th>
        <th>
          Date
        </th>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`renderSnapshots renderBloodButton 1`] = `
<div
  className="status"
>
  <p
    className="status-text"
  >
    Blood work schedule
  </p>
  <p
    className="status-number"
    id="bloodwork-button"
  >
    0
  </p>
</div>
`;

exports[`renderSnapshots renderBloodworkDetailsTable 1`] = `<bloodworkDetailsTable />`;

exports[`renderSnapshots renderBloodworkScheduleTable 1`] = `<bloodworkScheduleTable />`;

exports[`renderSnapshots renderCollectPrescription 1`] = `
<CollectPrescription
  classes={
    Object {
      "button": "CollectPrescription-button-1",
    }
  }
/>
`;

exports[`renderSnapshots renderMedButton 1`] = `
<div
  className="status"
>
  <p
    className="status-text"
  >
    Medication Pickup
  </p>
  <p
    className="status-number"
    id="medication-button"
  >
    0
  </p>
</div>
`;

exports[`renderSnapshots renderSidebar 1`] = `
<div
  className="sidebar"
>
  <div
    className="title"
  >
    <img
      alt="Pharmacy One logo"
      className="logo"
      src="p1logo.png"
    />
    <h1>
      PharmacyOne
    </h1>
  </div>
  <ul
    className="sidebarList"
  >
    <li
      className="row"
      id=""
      key="0"
      onClick={[Function]}
    >
      <div
        id="icon"
      >
        <Memo([object Object]) />
      </div>
      <div
        id="title"
      >
        Home
      </div>
    </li>
    <li
      className="row"
      id=""
      key="1"
      onClick={[Function]}
    >
      <div
        id="icon"
      >
        <Memo([object Object]) />
      </div>
      <div
        id="title"
      >
        Blood Work
      </div>
    </li>
    <li
      className="row"
      id=""
      key="2"
      onClick={[Function]}
    >
      <div
        id="icon"
      >
        <Memo([object Object]) />
      </div>
      <div
        id="title"
      >
        Medication
      </div>
    </li>
    <li
      className="row"
      id=""
      key="3"
      onClick={[Function]}
    >
      <div
        id="icon"
      >
        <Memo([object Object]) />
      </div>
      <div
        id="title"
      >
        Prescriptions
      </div>
    </li>
    <li
      className="row"
      id=""
      key="4"
      onClick={[Function]}
    >
      <div
        id="icon"
      >
        <Memo([object Object]) />
      </div>
      <div
        id="title"
      >
        Patient
      </div>
    </li>
    <li
      className="row"
      id="active"
      key="5"
      onClick={[Function]}
    >
      <div
        id="icon"
      >
        <Memo([object Object]) />
      </div>
      <div
        id="title"
      >
        Log Out
      </div>
    </li>
  </ul>
</div>
`;

exports[`renderSnapshots rendersPatientTable 1`] = `
<div>
  <table
    className="table"
    id="patients-table"
  >
    <tbody>
      <tr>
        <th>
          NHS ID
        </th>
        <th>
          Name
        </th>
        <th>
          Email
        </th>
        <th>
          Mobile
        </th>
        <th>
          Bloodwork Valid?
        </th>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`renderSnapshots rendersPrescriptionTable 1`] = `
<div>
  <h2>
    Prescriptions for pickup
  </h2>
  <table
    className="table"
    id="prescription-table"
  >
    <tbody>
      <tr>
        <th>
          Prescription No.
        </th>
        <th>
          Name
        </th>
        <th>
          NHS ID
        </th>
        <th>
          Medication
        </th>
        <th>
          Date
        </th>
      </tr>
    </tbody>
  </table>
</div>
`;
